mmengine/runner/runner.py

```diff
@@ -464,7 +476,8 @@ def from_cfg(cls, cfg: ConfigType) -> 'Runner':
     @property
     def experiment_name(self):
         """str: Name of experiment."""
-        return self._experiment_name
+        return f'{self._experiment_name}_{self._timestamp}' if \
+            self._experiment_name is not None else self._timestamp
```

Comment:
Why do we need to make `self.experiment_name` different from `self._self.experiment_name`

Reply:
See #614 , we have to keep then name and id for vis backends.

Reply:
I don't think it is an elegant way to extend the signature of `BaseVisBackend` for a specific backend. In other words, if [Neptune](https://neptune.ai/?utm_source=googleads&utm_medium=googleads&utm_campaign=brand_ads&gclid=Cj0KCQjw48OaBhDWARIsAMd966CBVg8JyQcHQSMGuBAzCwsLxTZxQSREyV0hsqottHdnsgQqoQ4LjVgaAqHbEALw_wcB) require another argument, we still need to change the interface. 

I think a workaround could be like that:

- Make `BaseVisBackend` accept **kwargs arguments.
- Set `experiment` argument in 
https://github.com/open-mmlab/mmengine/blob/dc01545e267f5bdf5afa2e1fdf781eb05f7df12c/mmengine/runner/runner.py#L391
- Parse `experiment_name` and `experiment_id` through `runner.experiment` in `WandbVisBackend`.

We may invite @zhouzaida @hhaAndroid to have look and ask for their opinion.

Reply:
In fact, all the Software as a Service(SaaS) Logger visbackends will take the `id` argument like wandb. See
- [Comet ML](https://www.comet.com/docs/v2/api-and-sdk/python-sdk/reference/Experiment/#:~:text=actually%20logging%20anything.-,experiment_key,-%3A%20Optional.%20If%20provided)
- [MLflow](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.start_run)
- [Neptune](https://docs.neptune.ai/api/neptune/#init_run)
and [Wandb](https://docs.wandb.ai/ref/python/init#:~:text=this%20integration.-,id,-(str%2C%20optional)%20A).

That's why I think add the `id` param is better. I agree with you for the `**kwargs`, but as the `runner.experiment`, I think `name` and `id` will be better, because we will use them for many loggers (if we want to add more SaaS loggers). The single `runner.experiment` may cause mistake unnecessary.

But, if you persist in the `runner.experiment` way, I can go through with it.

Reply:
Alright, I've understood. Since it is a common practice, It sounds proper to add an `experiment_name` and ``experiment`_id` argument to `BaseVisBackend`. @hhaAndroid @zhouzaida